core/domain/learner_playlist_services_test.py

```diff
@@ -100,162 +101,128 @@ def _get_all_learner_playlist_collection_ids(self, user_id):
-        # added to the learner playlist.
-        learner_progress_services.add_exp_to_learner_playlist(
-            self.user_id, self.EXP_ID_3)
-        self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(
-                self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
-
-        # Empty the learner playlist.
-        learner_playlist_services.remove_exploration_from_learner_playlist(
-            self.user_id, self.EXP_ID_0)
-        learner_playlist_services.remove_exploration_from_learner_playlist(
-            self.user_id, self.EXP_ID_1)
-        self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(
-                self.user_id), [])
-
-        # Test that the length of the learner playlist doesn't exceed
-        # MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT.
-        # List of explorations to be added.
+    def test_nunmber_of_explorations_cannot_exceed_max(self):
+        # Add MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT explorations.
         exp_ids = ['SAMPLE_EXP_ID_%s' % index for index in range(
             0, MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT)]
         for exp_id in exp_ids:
             learner_progress_services.add_exp_to_learner_playlist(
                 self.user_id, exp_id)
         self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(
-                self.user_id), exp_ids)
+            self._get_all_learner_playlist_exp_ids(self.user_id), exp_ids)
 
-        # Now if we try adding another exploration, it shouldn't be added as the
-        # list length would exceed MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT.
+        # Now if we try to add another exploration at the end of the list,
+        # it shouldn't be added as the list length would exceed
+        # MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT.
         learner_playlist_services.mark_exploration_to_be_played_later(
-            self.user_id, 'SAMPLE_EXP_ID')
+            self.user_id, 'SAPMLE_EXP_ID',
+            position_to_be_inserted=MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT)
+        self.assertEqual(
+            self._get_all_learner_playlist_exp_ids(self.user_id), exp_ids)
 
-        # The list still remains the same.
+        # Also if we try adding another exploration at no specific location,
+        # it shouldn't be added as the list length would exceed
+        # MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT.
+        learner_playlist_services.mark_exploration_to_be_played_later(
+            self.user_id, 'SAMPLE_EXP_ID')
         self.assertEqual(
-            self._get_all_learner_playlist_exp_ids(
-                self.user_id), exp_ids)
+            self._get_all_learner_playlist_exp_ids(self.user_id), exp_ids)
+
+    def test_adding_subscribed_collections_to_playlist_does_not_add_again(self):
+        # Subscribe to collection.
+        subscription_services.subscribe_to_collection(
+            self.user_id, 'SUBSCRIBED_ID')
 
-    def test_mark_collection_to_be_played_later(self):
```

Comment:
Split this into multiple test so that each function checks a single behaviour.

Reply:
Got it. Thanks!